### PR TITLE
MTP-901 - Search - Prisoners

### PR DESCRIPTION
**Pull request by gavinwye**
First, pass on the following pages

## Global
Reordered primary nav to reflect most visited pages. Last month unique visits to each of the page were:
   - prisoners, 1038
   - senders, 567
   - credits, 416
   - disbursements, 116 (not available in all prisons)

## Prisoners

- removed links from tables with the exception of the view details link
- increased the size of the search form label
- increased the size of the search filed and simplified button layout
- moved search again button to the bottom of the results page
  - _currently thinking about duplicating this at the top too_
- added pagination to the page but it's not working yet
- increased the font size of the table from 14 to 16

### CODE REVIEW

File: app/views/prisoners/index.html [hunk 1]
```diff
@@ -15,18 +15,38 @@ <h1 class="govuk-heading-xl govuk-!-margin-bottom-6">
...
+        <label class="govuk-label govuk-body govuk-!-font-size-27" for="width-20">
+          Search prisoner number or name
         </label>
+        <span id="event-name-hint" class="govuk-hint">
+          Prison numbers should be in the format 'A1234AB'.
+        </span>
         <input class="govuk-input govuk-!-width-two-thirds" id="prisonerSearch" name="prisonerSearch" type="text">
         <button type="submit" class="govuk-button">
-          Search
+          Search prisoners
```
Comment (gavinwye) [inline app/views/prisoners/index.html]: Why have you done this??

Comment (gavinwye) [inline app/views/prisoners/index.html]: ,m ,m

**Comment by gavinwye** [comment]
**TODO**
- update pagination for results 
- make search terms bold

**Comment by gavinwye** [comment]
## Playing with the order of content on the senders page

image

**Comment by RoseFaulkner** [comment]
I like the search and the tabbed credits/disbursements and I remember positive comments on this from UR. Standardising 'view details' across the sections makes sense.

For the basic search, do we know for 'prisoners' that prisoner number and name is what is most commonly searched? I wondered how common it is for a user to search across prisons and whether it's the right decision to bury that in 'Advanced'? And there's no indication of the prison/prisons they're currently in, I thought we were going to show that as we do with the live version? I think we need that to orientate the user. 

I'm not sure about changing 'payment source' to 'senders'. I understand the human side of it, everyone is a sender. And language wise, that feels to make sense at first. But as far as I'm aware, there's no UR insight to support this and I think it might be misleading. I think we should understand what language makes best sense before changing this. 

The white tab in the top nav that highlights the section you're in is missing. Assume this is just because it's the prototype and it hasn't been removed?

File: app/views/prisoners/index.html [hunk 2]
```diff
@@ -7,26 +7,32 @@
...
-        </button>
-        <p><a href="/advanced-search/" class="govuk-link">Advanced search</a></p>
-      </div>
+      {{ govukInput({
+        label: {
+          classes: "govuk-body govuk-!-font-size-27",
+          text: "Search prisoner number or name"
+        },
+        hint: {
+          text: "Prisoner numbers should be in the format 'A1234AB'."
```
Comment (janestead) [inline app/views/prisoners/index.html]: Line 23
Replace with

For example, name or A1234BC

File: app/views/prisoners/results.html [hunk 3]
```diff
@@ -68,8 +59,9 @@ <h2 class="govuk-heading-l">Credits</h2>
 
       </section>
       <section class="govuk-tabs__panel govuk-tabs__panel--hidden" id="disbursements">
-        <h2 class="govuk-heading-l">Disbursements</h2>
-        <table class="govuk-table govuk-!-font-size-14 ">
+        <!-- <h2 class="govuk-heading-l">Disbursements contaning "{{ data['prisonerSearch'] }}"</h2> -->
+        <table class="govuk-table govuk-!-font-size-16 ">
+          <!-- <p class="govuk-body">Disbursements contaning "{{ data['prisonerSearch'] }}"</p> -->
```
Comment (janestead) [inline app/views/prisoners/results.html]: Disbursements containing
Put 'i' in containing

File: app/views/prisoners/results.html [hunk 4]
```diff
@@ -52,9 +42,10 @@ <h2 class="govuk-tabs__title">
       </ul>
 
       <section class="govuk-tabs__panel" id="credits">
-        <h2 class="govuk-heading-l">Credits</h2>
-        <table class="govuk-table govuk-!-font-size-14 ">
-          <!-- <caption class="govuk-table__caption">Credits for your prisons sorted by date</caption> -->
+        <!-- <h2 class="govuk-heading-l">Credits contaning "{{ data['prisonerSearch'] }}"</h2> -->
+        <table class="govuk-table govuk-!-font-size-16 ">
+          <!-- <p class="govuk-body">Credits contaning "{{ data['prisonerSearch'] }}"</p> -->
+          <!-- <caption class="govuk-table__caption">Credits contaning "{{ data['prisonerSearch'] }}"</caption> -->
```
Comment (janestead) [inline app/views/prisoners/results.html]: 'i' in containing

File: app/views/prisoners/results.html [hunk 5]
```diff
@@ -7,32 +7,22 @@
...
-        <button type="submit" class="govuk-button">
-          Search
-        </button>
-        <p><a href="/advanced-search/" class="govuk-link">Advanced search</a></p>
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-one-half">
+        {%- if prisoners.length == 0 -%}
+        <p class="govuk-body-l">Your search for <span class="govuk-!-font-weight-bold">"{{ data['prisonerSearch'] }}"</span> returned no results</p>
+        {%- else -%}
+        <p class="govuk-body-l">1 result containing prisoner number or name <span class="govuk-!-font-weight-bold">"{{ data['prisonerSearch'] }}"</span></p>
```
Comment (janestead) [inline app/views/prisoners/results.html]: Line should read:
Results containing 'XXXXX' 

(Do we need to say how many??)

Also, in 'Credits recieved' in the table, it's 'received', (e before i, after the c)

**Comment by janereid73** [comment]
@gavinwye @RoseFaulkner @janestead I've just performed a quick search from research transcripts and the term sender is more widely used by staff, in the same research reports there was no mention of payment sources. I didn't check prototype testing papers as people these could be biased by what people saw on the screen at the time. This quick search would suggest 'senders' is the term to use and not 'payment sources'. Here's some examples: https://docs.google.com/document/d/19CLkl-fvQ7Dh0Kifsqvp621adMtXgHAoRlckbrQcptE/edit

**Comment by gavinwye** [comment]
I've just added the results for payment sources in other prisons at the moment these display the 3 letter prison code but it would be better to display the full name but I'll have to update the json file to make this work.

**Comment by RoseFaulkner** [comment]
In the 'Payment sources' the copy is:
Search payment source name or email address
Common or incomplete names may show many results

but in 'Prisoners' the copy is:
Search prisoner number or name
For example, name or 'A1234BC'

In 'Prisoners' shouldn't we also put in the copy 'Common or incomplete names may show many results'?

**Comment by janestead** [comment]
Hang on, I haven't looked at this in its overall state yet... I should probably check it all first! I did it in bits so haven't done overall check yet.

**Pull request merged**